config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,22 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// apiIP holds the IP that can be used to contact the OpenShift API. It is the IP for a
+	// self-hosted load balancer in front of the API servers.
+	APIIP string `json:"apiIP"`
+
+	// ingressIP holds the IP for an ingress router that an external wildcard DNS record
+	// can point to for apps running within the BareMetal cluster.
+	IngressIP string `json:"ingressIP"`
```

Comment:
On this platform, the default ingresscontroller will be [exposed](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L83) using the [HostNetwork strategy](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L228).

The default ingress controller's domain is specified by the [cluster ingress configuration](https://github.com/openshift/api/blob/master/config/v1/types_ingress.go#L28).

Given those facts, is the following an accurate description of `IngressIP`?

>ingressIP is an external IP which routes to the default ingress controller.
>
>The IP is a suitable target of a wildcard DNS record used to resolve default route host names.

Reply:
Yes, that sounds right

Reply:
A note which might be useful going forward: one thing I would recommend is being careful referring to ingress controllers — we've eliminated the word "router" from the API in 4.x and the word "ingress" alone is fairly overloaded and often a source of confusion/ambiguity. For example, as stated, "ingress router" doesn't clearly map onto anything in the 4.x API (hence the use of "ingress controller" in my example description).